kuma/wiki/views/misc.py

```diff
@@ -0,0 +1,191 @@
+def autosuggest_documents(request):
+    """
+    Returns the closest title matches for front-end autosuggests
+    """
+    partial_title = request.GET.get('term', '')
+    locale = request.GET.get('locale', False)
+    current_locale = request.GET.get('current_locale', False)
+    exclude_current_locale = request.GET.get('exclude_current_locale', False)
+
+    if not partial_title:
+        # Only handle actual autosuggest requests, not requests for a
+        # memory-busting list of all documents.
+        return HttpResponseBadRequest(_lazy('Autosuggest requires a partial '
+                                            'title. For a full document '
+                                            'index, see the main page.'))
+
+    # Retrieve all documents that aren't redirects or templates
+    docs = (Document.objects.extra(select={'length': 'Length(slug)'})
+                            .filter(title__icontains=partial_title,
+                                    is_template=0,
+                                    is_redirect=0)
+                            .exclude(slug__icontains='Talk:')  # Remove old talk pages
+                            .order_by('title', 'length'))
+
+    # All locales are assumed, unless a specific locale is requested or banned
+    if locale:
+        docs = docs.filter(locale=locale)
+    if current_locale:
+        docs = docs.filter(locale=request.locale)
+    if exclude_current_locale:
+        docs = docs.exclude(locale=request.locale)
+
+    # Generates a list of acceptable docs
+    docs_list = []
+    for doc in docs:
+        data = doc.get_json_data()
+        data['label'] += ' [' + doc.locale + ']'
+        docs_list.append(data)
+
+    return JsonResponse(docs_list, safe=False)
+
+
+@xframe_options_sameorigin
+@process_document_path
+def flag(request, document_slug, document_locale):
+    """
+    Flag a document for something.
+    """
+    doc = get_object_or_404(Document,
+                            slug=document_slug,
+                            locale=document_locale)
+
+    if request.method == 'POST':
+        form = DocumentContentFlagForm(data=request.POST)
+        if form.is_valid():
+            flag_type = form.cleaned_data['flag_type']
+            recipients = None
+            if (flag_type in FLAG_NOTIFICATIONS and
+                    FLAG_NOTIFICATIONS[flag_type]):
+                query = Q(email__isnull=True) | Q(email='')
+                recipients = list(User.objects.exclude(query)
+                                              .values_list('email', flat=True))
+
+            flag, created = ContentFlag.objects.flag(
+                request=request, object=doc,
+                flag_type=flag_type,
+                explanation=form.cleaned_data['explanation'],
+                recipients=recipients)
+            return redirect(doc)
+    else:
+        form = DocumentContentFlagForm(data=request.GET)
+
+    return render(request, 'wiki/flag.html', {'form': form, 'doc': doc})
```

Comment:
Ah, https://github.com/mozilla/kuma/commit/b9f1a32cff26a1b3ecee7fbe6c0bcb5ac6afb394 added the view, but no template? @groovecoder do you remember any of this?